### DISCUSSION TITLE
How to release mouse right click ?

### DISCUSSION THREAD

**Discussion by hiro-k-phys** [pyvista/pyvista#5378] [Q&A] [answered]
Hello everyone,
I am testing the following example with a context menu, and by right clicking the actor, the context menu appears. However, after choosing "pick", it does not release the right click. Can anyone suggest how to release the right click?
Thank you so much in advance!

```
from PySide6.QtWidgets import QApplication, QVBoxLayout, QMenu, QDialog
from PySide6.QtGui import QCursor
import pyvista as pv
from pyvistaqt import QtInteractor

class QtPyvista(QDialog):
    def __init__(self):
        super().__init__()
        self.plotter = QtInteractor(parent=self)
        self.plotter.background_color = "white"
        self.plotter.enable_mesh_picking(
            self.pick, use_actor=True, show=False, show_message=False
        )
        self.plotter.add_mesh(pv.Cube(), name="actor")
        layout = QVBoxLayout(self)
        layout.addWidget(self.plotter)

    def pick(self, actor):
        menu = QMenu(self)
        action_remove = menu.addAction("pick")
        action_remove.triggered.connect(lambda: print(f"pick {actor.name}"))
        menu.exec(QCursor().pos())

app = QApplication([])
QtPyvista().show()
app.exec()
```

**Comment by banesullivan** [comment]
I'm wondering if you are still having this issue with the latest release: `0.43.1`? I made some changes to the mouse button release events in https://github.com/pyvista/pyvista/pull/5160 for https://github.com/pyvista/pyvista/issues/4976 and this sounds mighty similar.

**Reply by banesullivan** [reply]
Hm. I just tried this with 0.43.1 and no luck... I wonder if there is a similar issue going on with the QtInteractor? Not sure if I have the availability to dive into this, but maybe my PR linked above will give you a lead to investigate this

**Reply by hiro-k-phys** [reply]
Thank you for related information.
In the callback of enable_mesh_picking, it seems that no procedures have been done for releasing mouse click, and as a result it keeps holding a mouse, i.e., the functionality "continuously zoom the rendering scene" is continuing.
As I am not familiar to the internal structure of vtk and pyvista, I could not investigate this issue further.

**Comment by hiro-k-phys** [comment, marked as answer]
it is solve by explicitly releasing mouse button as follows.
just before `menu.exec(QCursor().pos())`, we insert the releasing code,
```
event = QMouseEvent(QEvent.MouseButtonRelease,QCursor().pos(),QCursor().pos(),Qt.LeftButton,Qt.LeftButton,Qt.NoModifier)
QCoreApplication.sendEvent(self.plotter, event)
```
I am not sure it is an appropriate way, but anyway, the mouse button is released.
